### DISCUSSION TITLE
`<select>`

### DISCUSSION THREAD

**Discussion by amitu** [fastn-stack/fastn#1329] [Ideas & RFCs]
Maybe including `<hr>` as separator between items. Option Tree?

**Comment by harshdoesdev** [comment]
Maybe we can implement the native <select> element itself?

```ftd
-- ftd.select:
options: Sunday, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday
selected: Monday
placeholder: Please select a day
$on-select$: $handle-select($v = $VALUE)
```

Output:
```ftd
<select>
<option selected disabled>Please select a day</option>                  // this is because select does not supports placeholder by default
<option>Sunday</option>
<option>Monday</option>
<option>Tuesday</option>
...
</select>
```

**Reply by amitu** [reply]
Why is the preview broken here!? Github bug it seems.

**Reply by amitu** [reply]
One: Options have to allow id (to be set to variable) and "human display string" to be shown. 

Two: Let's also not do the ftd.text-input mistake, and pass a mutable variable to the component so we do not have to write event handler, and variable automatically gets set. In all such event handlers we mutate the variable anyways, so directly passing mutable variable is better. 

Three: On-Select, let's use `on-change` for every input. No need to have a unique name for each. on-change is equally clear and I do not have to remember which input uses what event handler name. 

Four: why would select have a placeholder?
